weavedns/main.go

```diff
@@ -127,8 +127,13 @@ func main() {
 		Error.Fatal("[main] Failed to initialize the WeaveDNS server", err)
 	}
 
+	httpListener, err := net.Listen("tcp", fmt.Sprintf(":%d", httpPort))
+	if err != nil {
+		Error.Fatal("[main] Unable to create http listener: ", err)
+	}
+
 	go SignalHandlerLoop(srv)
-	go weavedns.ListenHTTP(version, srv, domain, zone, httpPort)
+	go weavedns.ServeHTTP(httpListener, version, srv, domain, zone)
```

Comment:
I think the `main()` function is not the place for initializing low level things like a TCP listener. Following the same interface provided by [the standard library](http://golang.org/pkg/net/http/), I think there should be a [`ListenAndServe()`](http://golang.org/pkg/net/http/#ListenAndServe) that creates the listener automatically, and a [`Serve()`](http://golang.org/pkg/net/http/#Serve) where you can provide your own listener...